analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -113,20 +112,20 @@ public sealed class SetLocaleForDataTypes : SonarDiagnosticAnalyzer
                 ? semanticModel.GetSymbolInfo(objectCreation.Type).Symbol
                 : semanticModel.GetSymbolInfo(constructorCall).Symbol?.ContainingType;
 
-        private static SyntaxNode GetAssignmentTargetVariable(SyntaxNode objectCreation) =>
+        private static ImmutableArray<SyntaxNode> GetAssignmentTargetVariables(SyntaxNode objectCreation) =>
             objectCreation.GetFirstNonParenthesizedParent() switch
             {
-                AssignmentExpressionSyntax assignment => assignment.Left,
-                EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration => variableDeclaration.Variables.Last(),
-                _ => null
+                AssignmentExpressionSyntax assignment => assignment.AssignmentTargets(),
+                EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration =>
+                    variableDeclaration.Variables.Select(x => (SyntaxNode)x).ToImmutableArray(),
+                ArgumentSyntax argument when argument.Ancestors().OfType<AssignmentExpressionSyntax>().FirstOrDefault() is { } expressionSyntax =>
+                    expressionSyntax.AssignmentTargets(),
+                _ => ImmutableArray<SyntaxNode>.Empty
```

Comment:
I think you should use `FindAssignmentComplement` here. With that change, `private static ImmutableArray<SyntaxNode> GetAssignmentTargetVariables(SyntaxNode objectCreation)` can be changed backed to return a single SyntaxNode. There are some obscure patterns, that allow you to can assign to more than one variable, but these are out of scope of the PR. e.g.

```cs
        object i, j;
        i = (j = new object()); // i and j point to the same value now
        var k = (j ??= new object()); // This is a bit more popular recently for caching.
```